### DISCUSSION TITLE
Build problems with mozjpeg binaries

### DISCUSSION THREAD

**Discussion by andrews05** [libvips/libvips#2138] [Q&A]
I'm using the mozjpeg binaries from [here](https://mozjpeg.codelove.de/binaries.html) and have run into a problem. I'm configuring with `./configure  --with-jpeg-includes=/opt/mozjpeg/include --with-jpeg-libraries=/opt/mozjpeg/lib64` and everything successfully builds but running it gives an error:
`vipsthumbnail: error while loading shared libraries: libjpeg.so.62: cannot open shared object file: No such file or directory`
It does work if I build mozjpeg manually but I was hoping to avoid having to do that. I've tried both the 3.3.1 and 4.0 binaries and same issue.
I'm guessing issue is probably with the mozjpeg binaries but my understanding of libaries and linking is limited so thought I'd ask for here help.

**Comment by jcupitt** [comment]
Hi, at a guess you need to add `/opt/mozjpeg/lib64` to your library path. I would check `/etc/ld.so.conf.d`, or alternatively set `LD_LIBRARY_PATH`.

**Reply by andrews05** [reply]
Is that relevant for running or for building? Ideally I'd want to build it in such a way that the library path isn't needed at runtime. Like how it works if I build mozjpeg manually, which doesn't appear to use either of those.

**Reply by andrews05** [reply]
Alight, I fixed it by setting LD_RUN_PATH before building vips!
Thanks for your comment though, that led me to this solution.
